### ISSUE TITLE
What should happen when Task::local use detach()

### ISSUE THREAD

**Issue by attliaLin** [stjepang/smol#93]
```rust

use smol::Task;

fn main() {
    smol::run(async {
        Task::local(async {
            print!("helloworld");
        })
        .detach();
    });
}

```

Nothing happened.

I expect it will get the `helloworld` or tell me it's not allowed.

**Comment by laizy** [comment]
@attliaLin `detach` will let the task running in the background. when `smol::run` finished, it will drop all tasks spawned using `Task::local`, see #76 . So you need make sure all local spawned tasks have completed before returning from `smol::run`.  the following code should print `helloworld`.

```rust
use smol::Task;
use smol::Timer;
use std::time::Duration;

fn main() {
    smol::run(async {
        Task::local(async {
            println!("helloworld");
        })
        .detach();
	Timer::after(Duration::from_secs(1)).await;
    });
```

**Comment by attliaLin** [comment]
> @attliaLin `detach` will let the task running in the background. when `smol::run` finished, it will drop all tasks spawned using `Task::local`, see #76 . So you need make sure all local spawned tasks have completed before returning from `smol::run`. the following code should print `helloworld`.
> 
> ```rust
> use smol::Task;
> use smol::Timer;
> use std::time::Duration;
> 
> fn main() {
>     smol::run(async {
>         Task::local(async {
>             println!("helloworld");
>         })
>         .detach();
> 	Timer::after(Duration::from_secs(1)).await;
>     });
> ```

It works! Thx!
